realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -384,8 +396,10 @@ bool BaseRealSenseNode::get_coords_cb(realsense2_camera_srvs::srv::CoordinateReq
                 // an Eigen vector is created to easily apply spatial transforms, however the coords are published in the camera frame for now
                 Eigen::Vector3f v(point_requested_coords.x, point_requested_coords.y, point_requested_coords.z);
                 Eigen::Matrix3f m;
+                Eigen::Translation<float,3> t;
                 m = Eigen::AngleAxisf(-M_PI_2, Eigen::Vector3f::UnitZ())*Eigen::AngleAxisf((-90.0 -_cam_pitch)*M_PI/180, Eigen::Vector3f::UnitX());
-                v = m*v;
+                t = Eigen::Translation<float,3>(_camera_link_x, _camera_link_y, _camera_link_z);
+                v = t*m*v;
```

Comment:
I know this translation is the right way to do it. Nonetheless I think the semantic navigation node uses this but it wants the __camera_link__ coordinates, not the chassis frame (center of the robot), because the waypointcontroller I think takes control from the front reference of the robot. Let's discuss this with @dadaroce

Reply:
Thinking twice this is not strictly necessary. However in that case we should specify that `camera_link` should be the base frame for passing points to the `get_pixel` service that were obtained with the `get_coors` service.

Reply:
Agree